From 07d773ad0044fe3565c973733b4c00d0d9a01b14 Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Wed, 2 Jul 2025 14:04:51 +0530
Subject: [PATCH] fix(cli): enable help output by explicitly enabling clap's
 help feature

- In clap v4, the feature system was restructured to be fully granular
- With `default-features = false`, **no features** are enabled, including help
- Previously in v3, help output was always included by default
- Without explicitly enabling the `help` feature, running:
    ./linkerd-network-validator --help
  results in:
    error: unexpected argument found
- This commit adds the `help` feature back to enable standard CLI help text

Ref: https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#400---2022-09-28
---
 validator/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/validator/Cargo.toml b/validator/Cargo.toml
index 737bc8a4..d3537f97 100644
--- a/validator/Cargo.toml
+++ b/validator/Cargo.toml
@@ -16,7 +16,7 @@ tracing = "0.1"
 [dependencies.clap]
 version = "4"
 default-features = false
-features = ["derive", "env", "std"]
+features = ["derive", "env", "std", "help"]
 
 [dependencies.tokio]
 version = "1"
